src/composables/useAppLoadingNotifications.ts

```diff
@@ -0,0 +1,73 @@
+import { TYPE } from 'vue-toastification';
+import { useToast } from '@/src/composables/useToast';
+import { ToastID } from 'vue-toastification/dist/types/src/types';
+import { useMessageStore } from '../store/messages';
+
+export function useAppLoadingNotifications() {
+  const toast = useToast();
+  const messageStore = useMessageStore();
+
+  let loadingCount = 0;
+  let toastID: ToastID | null = null;
+  let error: Error | null = null;
+
+  const resetState = () => {
+    loadingCount = 0;
+    toastID = null;
+    error = null;
+  };
+
+  const _setError = (err: Error) => {
+    error = err;
+  };
+
+  const startLoading = () => {
+    loadingCount++;
+    if (toastID === null) {
+      toastID = toast.info('Loading datasets...', {
+        timeout: false,
+        closeButton: false,
+        closeOnClick: false,
+      });
+    }
+  };
+
+  const stopLoading = () => {
+    loadingCount--;
+    if (loadingCount === 0 && toastID !== null) {
+      if (error) {
+        toast.dismiss(toastID);
```

Comment:
Just for my learning: Why did you not throw a bang in `toast.dismiss(toastID!)` and take the runtime error? If `toastID === null` then startLoading would not have been called, which would be a silent bug now?

Reply:
Just defensive programming. Also, I have another project where the linter complains about non-null assertions, so I likely implicitly followed that here.